packages/react/menu/src/Menu.tsx

```diff
@@ -645,6 +652,16 @@ function getCheckedState(checked: boolean) {
   return checked ? 'checked' : 'unchecked';
 }
 
+function focusFirst(candidates: HTMLElement[]) {
+  const PREVIOUSLY_FOCUSED_ELEMENT = document.activeElement;
+  for (const candidate of candidates) {
+    // if focus is already where we want to go, we don't want to keep going through the candidates
+    if (candidate === PREVIOUSLY_FOCUSED_ELEMENT) return;
+    candidate.focus();
+    if (document.activeElement !== PREVIOUSLY_FOCUSED_ELEMENT) return;
+  }
+}
+
```

Comment:
This is duplicated from `RovingFocusGroup`, let me know if you think we should do something about that.